src/pull.go

```diff
@@ -51,8 +52,47 @@ func (g *Commands) Pull(byId bool) error {
 	cl, clashes, err := pullLikeResolve(g, byId)
 
 	if len(clashes) >= 1 {
-		warnClashesPersist(g.log, clashes)
-		return ErrClashesDetected
+		if g.opts.FixClashes {
+			g.log.Logln("Some clashes found")
+			if g.opts.canPrompt() {
+				if !promptForChanges("Try fixing them [Y/N] ? ") {
+					return ErrClashesDetected
+				}
+			}
+			sort.Sort(ByPath(clashes))
+
+			lastPath := ""
+			i := 0
+			for _, v := range clashes {
+				if v.Path != lastPath {
+					lastPath = v.Path
+					i = 2
+					continue
+				}
+
+				var newPath string
+				for {
+					ext := filepath.Ext(v.Path)
+					name := strings.TrimSuffix(v.Path, ext)
+					newPath = fmt.Sprintf("%v_%d%v", name, i, ext)
+					i++
+
+					var dupCheck *File
+					dupCheck, err = g.rem.FindByPath(newPath)
+					if err != nil && dupCheck == nil {
+						newPath = filepath.Base(newPath)
+						break
+					}
+				}
+
+				g.log.Logln(fmt.Sprintf("%v %v -> %v", v.Path, v.Src.Id, newPath))
+				g.rem.rename(v.Src.Id, newPath)
+			}
+			return fmt.Errorf("Clashes were fixed, try pulling again.")
+		} else {
```

Comment:
Could we switch up the conditionals so instead of
```go
if g.opts.FixClashes {
    ...
} else {
    warnClashesPersist(g.log, clashes)
    return ErrClashesDetected
}
```

let's make it

```go
if !g.opts.FixClashes {
    warnClashesPersist(g.log, clashes)
    return ErrClashesDetected
} else {
    ....
}
```